aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -109,44 +126,98 @@ module aptos_framework::transaction_validation {
     fun prologue_common(
         sender: &signer,
         gas_payer: &signer,
-        txn_sequence_number: u64,
+        replay_protector: ReplayProtector,
         txn_authentication_key: Option<vector<u8>>,
         txn_gas_price: u64,
         txn_max_gas_units: u64,
         txn_expiration_time: u64,
         chain_id: u8,
         is_simulation: bool,
     ) {
+        let sender_address = signer::address_of(sender);
+        let gas_payer_address = signer::address_of(gas_payer);
         assert!(
             timestamp::now_seconds() < txn_expiration_time,
             error::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED),
         );
         assert!(chain_id::get() == chain_id, error::invalid_argument(PROLOGUE_EBAD_CHAIN_ID));
 
+        // TODO[Orderless]: Here, we are mantaining the same order of validation steps as before orderless txns were introduced.
+        // Ideally, do the replay protection check in the end after the authentication key check and gas payment checks.
```

Comment:
the code contradicts with what the comment says? the check is before authentication and gas check

Reply:
My bad. Switched the order of authentication and replay protection checks.